core/src/main/java/com/nisovin/magicspells/util/Util.java

```diff
@@ -66,32 +66,32 @@ public static Material getMaterial(String name) {
 		return Material.matchMaterial(name);
 	}
 
-	// - <potionEffectType> (level) (duration) (ambient)
-	public static PotionEffect buildPotionEffect(String effectString) {
+	// - <potionEffectType> (duration) (amplifier) (ambient) (particles) (icon)
+	public static PotionEffect buildPotionEffect(String effectString, boolean durationThenAmplifier) {
```

Comment:
Rather than confusingly support 2 formats that are exactly the same, but with 2 values swapped, perhaps deprecate both options & add a `potion-effects` section to `EntityData`?

Reply:
Right now, that format (now method too) is used in: `MinionSpell`, `SpawnEntitySpell`, `ParticleCloudSpell`, and both magic item parsers. I agree that most options in the first two spells should be moved to `EntityData`, but the spell options override entity data at the moment, so we'll have to deal with most of those spells having breaking changes at some other point in time. We could even hide away Minion behaviour with mob goals and unify all three spells in an entity spawning spell?

I agree with your suggestion, but my commit is mostly a refactor, and the suggestion requires breaking changes scheduled for another time. We just know the magic item format will be abandoned.

Reply:
Reverted `MinionSpell` and `SpawnEntitySpell` to avoid expanding the format. I added `potion-effects` to `EntityData`, but Minion does not use entity data, so we'll deal with it and its potions when we deal with buff spells. `ParticleCloudSpell` now uses `Util#getPotionEffects` with the minor breaking change of `particles` and `ambient` being swapped.